src/werkzeug/datastructures/cache_control.py

```diff
@@ -172,26 +234,38 @@ class ResponseCacheControl(_CacheControl):
     for that class.
 
     .. versionchanged:: 3.1
-       ``no_transform`` is a boolean when present.
+        Dict values are always ``str | None``. Setting properties will
+        convert the value to a string. Setting a non-bool property to
+        ``False`` is equivalent to setting it to ``None``. Getting typed
+        properties will return ``None`` if conversion raises
+        ``ValueError``, rather than the string.
+
+    .. versionchanged:: 3.1
+        ``private`` is a boolean, it is ``False`` instead of ``"*"``
+        when not present.
```

Comment:
It returned `*` when *present* without an argument. It should not be bool since arguments [are permitted](https://httpwg.org/specs/rfc9111.html#cache-response-directive.private):

3.0.6:
```python
>>> assert parse_cache_control_header("", cls=ResponseCacheControl).private is None
>>> assert parse_cache_control_header("private", cls=ResponseCacheControl).private == '*'
>>> assert parse_cache_control_header('private="Set-Cookie, Authorization"', cls=ResponseCacheControl).private == 'Set-Cookie, Authorization'
```

main:
```python
>>> assert parse_cache_control_header("", cls=ResponseCacheControl).private is False
>>> assert parse_cache_control_header("private", cls=ResponseCacheControl).private is True
>>> cc = parse_cache_control_header('private="Set-Cookie, Authorization"', cls=ResponseCacheControl)
>>> assert cc.private == 'Set-Cookie, Authorization', f"{cc.private=}"
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError: cc.private=True
```

Reply:
Totally right, I missed that it can have an argument. It's because I was toggling between the spec and MDN, which doesn't mention the argument https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control.